docs/local-lxd/explanation/charm-architecture.md

```diff
@@ -0,0 +1,132 @@
+The GitHub self-hosted runner applications will be configured to utilise the proxy configuration. 
+This involves setting environment variables such as `http_proxy`, `https_proxy`, `no_proxy`, `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY`
+in various locations within the runner environment, such as `/etc/environment`.
+
+However, employing this approach with environment variables has its drawbacks. 
+Not all applications within a workflow may adhere to these variables as they 
+[lack standardisation](https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/). 
+This inconsistency can result in failed workflows, prompting the introduction of aproxy, as detailed in the subsection below.
+
+### aproxy
+If the proxy configuration is utilised and [aproxy](https://github.com/canonical/aproxy) is specified through the charm's configuration option,
+all HTTP(S) requests to standard ports (80, 443) within the GitHub workflow will be automatically directed 
+to the specified HTTP(s) proxy. Network traffic destined for ports 80 and 443 is redirected to aproxy using iptables.
+aproxy then forwards received packets to the designated HTTP(S) proxy. 
+Beyond that, the environment variables (`http_proxy`, `https_proxy`, `no_proxy`, `HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY`)
+will no longer be defined  in the runner environment. 
+It's worth noting that this setup deviates from the behaviour when not using aproxy, 
+where these variables are set in the runner environment. In that scenario, traffic to non-standard ports 
+would also be directed to the HTTP(s) proxy, unlike when using aproxy.
+
+### denylist
+
+The nftables on the Juju machine are configured to deny traffic from the runner virtual machine to IPs on the [`denylist` configuration](https://charmhub.io/github-runner/configure#denylist). The runner will always have access to essential services such as DHCP and DNS, regardless of the denylist configuration.
+
+## GitHub API usage
+
+The charm requires a GitHub personal access token for the [`token` configuration](https://charmhub.io/github-runner/configure#token). This token is used for:
+
+- Requesting self-hosted runner registration tokens
+- Requesting self-hosted runner removal tokens
+- Requesting a list of runner applications
+- Requesting a list of self-hosted runners configured in an organization or repository
+- Deleting self-hosted runners
+
+The token is also passed to [repo-policy-compliance](https://github.com/canonical/repo-policy-compliance) to access GitHub API for the service.
+
+Note that the GitHub API uses a [rate-limiting mechanism](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28). When this is reached, the charm may not be able to perform the necessary operations and may go into
```

Comment:
```suggestion
Note that the GitHub API uses a [rate-limiting mechanism](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28). When this limit is reached, the charm may not be able to perform the necessary operations and may go into
```
Trying to make it clearer what "this" refers to, I think it's the rate limit but please clarify if I misunderstood!

Reply:
If you do not mind, as this document was not updated in thie PR, I will leave this improvements for future PRs (as otherwise this PR will become unwieldy)